src/app/Dashboard/DraggableRef.tsx

```diff
@@ -0,0 +1,145 @@
+ * one is included with the Software (each a "Larger Work" to which the Software
+ * is contributed by such licensors),
+ *
+ * without restriction, including without limitation the rights to copy, create
+ * derivative works of, display, perform, and distribute the Software and make,
+ * use, sell, offer for sale, import, export, have made, and have sold the
+ * Software and the Larger Work(s), and to sublicense the foregoing rights on
+ * either these or other terms.
+ *
+ * This license is subject to the following condition:
+ * The above copyright notice and either this complete permission notice or at
+ * a minimum a reference to the UPL must be included in all copies or
+ * substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+import {
+  CardConfig,
+  dashboardCardConfigResizeCardIntent,
+} from '@app/Shared/Redux/Configurations/DashboardConfigSlicer';
+import { RootState } from '@app/Shared/Redux/ReduxStore';
+import { gridSpans } from '@patternfly/react-core';
+import _ from 'lodash';
+import React from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { DashboardCardContext } from './ResizableCard';
+
+export interface DraggableRefProps {
+  dashboardId: number;
+  minimumSpan: gridSpans;
+}
+
+function normalizeAsGridSpans(val: number, min: number, max: number, a: gridSpans, b: gridSpans): gridSpans {
+  if (val < min) val = min;
+  else if (val > max) val = max;
+  const ans = Math.round((b - a) * ((val - min) / (max - min)) + a);
+  return _.clamp(ans, a, b) as gridSpans;
+}
+
+export const DraggableRef: React.FunctionComponent<DraggableRefProps> = ({
+  dashboardId,
+  minimumSpan,
+  ..._props
+}: DraggableRefProps) => {
+  const cardConfigs: CardConfig[] = useSelector((state: RootState) => state.dashboardConfigs.list);
+  const dispatch = useDispatch();
+
+  const cardRef = React.useContext(DashboardCardContext);
+  const isResizing = React.useRef<boolean>(false);
+  const setInitialVals = React.useRef<boolean>(true);
+  const minWidth = React.useRef<number>(0);
+
+  const callbackMouseMove = React.useCallback(
+    (e: MouseEvent) => {
+      const mousePos = e.clientX;
+      e.stopPropagation();
+      if (!isResizing.current) {
+        return;
+      }
+      if (cardRef.current && window.visualViewport) {
+        cardRef.current.scrollIntoView({ behavior: 'smooth', block: 'center', inline: 'end' });
+        const cardRect = cardRef.current.getBoundingClientRect();
+
+        if (setInitialVals.current) {
```

Comment:
Do you think we could use only `minWidth` ref but with declaration as:

```ts
const minWidth = React.useRef<number | undefined>(0);
```

Then, we just need to check if this minWith is defined, set if not and reset when done?